components/eam/src/physics/crm/physpkg.F90

```diff
@@ -1,1802 +1,1184 @@
+  use prescribed_ozone,         only: prescribed_ozone_register
+  use prescribed_volcaero,      only: prescribed_volcaero_register
+  use prescribed_aero,          only: prescribed_aero_register
+  use prescribed_ghg,           only: prescribed_ghg_register
+  use sslt_rebin,               only: sslt_rebin_register
+  use aoa_tracers,              only: aoa_tracers_register
+  use aircraft_emit,            only: aircraft_emit_register
+  use cam_diagnostics,          only: diag_register
+  use cospsimulator_intr,       only: cospsimulator_intr_register
+  use rad_constituents,         only: rad_cnst_get_info ! Added to query if it is a modal aero sim or not
+  use output_aerocom_aie,       only: output_aerocom_aie_register, do_aerocom_ind3
+  use crm_physics,              only: crm_physics_register
+  use vertical_diffusion,       only: vertical_diffusion_register
+  use cloud_diagnostics,        only: cloud_diagnostics_register
+  !---------------------------------------------------------------------------
+  ! Local variables
+  !---------------------------------------------------------------------------
+  integer :: m,mm     ! loop and constituent indices
+  integer :: dummy    ! for unused output from pbuf_add_field calls
+  integer :: nmodes
+  character(len=16) :: MMF_microphysics_scheme
+  !---------------------------------------------------------------------------
+  !---------------------------------------------------------------------------
+
+  call phys_getopts(do_aerocom_ind3_out         = do_aerocom_ind3,  &
+                    state_debug_checks_out      = state_debug_checks, &
+                    micro_do_icesupersat_out    = micro_do_icesupersat, &
+                    pergro_test_active_out      = pergro_test_active, &
+                    pergro_mods_out             = pergro_mods, &
+                    MMF_microphysics_scheme_out = MMF_microphysics_scheme )
+
+  ! Initialize dyn_time_lvls
+  call pbuf_init_time()
+
+  ! Register water vapor.
+  ! This must be the first call to cnst_add so that water vapor is constituent 1.
+  call cnst_add('Q', mwh2o, cpwv, 1.E-12_r8, mm, longname='Specific humidity', readiv=.true., is_convtran1=.true.)
+
+  ! Fields for physics package diagnostics
+  call pbuf_add_field('TINI',      'physpkg', dtype_r8, (/pcols,pver/), tini_idx)
+  call pbuf_add_field('QINI',      'physpkg', dtype_r8, (/pcols,pver/), qini_idx)
+  call pbuf_add_field('CLDLIQINI', 'physpkg', dtype_r8, (/pcols,pver/), cldliqini_idx)
+  call pbuf_add_field('CLDICEINI', 'physpkg', dtype_r8, (/pcols,pver/), cldiceini_idx)
+  call pbuf_add_field('static_ener_ac', 'global', dtype_r8, (/pcols/), static_ener_ac_idx)
+  call pbuf_add_field('water_vap_ac',   'global', dtype_r8, (/pcols/), water_vap_ac_idx)
+
+  ! check energy package
+  call check_energy_register
+
+  ! register fluxes for saving across time
+  if (phys_do_flux_avg()) call flux_avg_register()
+     
+  ! Some of these pbuf fields may not be needed for MMF
```

Comment:
Shouldn't we remove the ones that are not needed for MMF?

Reply:
They are used by cam_diagnostics.F90, except for the "STR" ones, I think those are the only ones I can delete. I'll modify the comment.